[DOCS] Adds Google Vertex AI tutorial (#112339)
elastic/elasticsearch: docs/reference/search/search-your-data/semantic-search-inference.asciidoc

IMPORTANT: For the easiest way to perform semantic search in the {stack}, refer to the <<semantic-search-semantic-text, `semantic_text`>> end-to-end tutorial.

The following examples use the:

* `embed-english-v3.0` model for https://docs.cohere.com/docs/cohere-embed[Cohere]
* `all-mpnet-base-v2` model from https://huggingface.co/sentence-transformers/all-mpnet-base-v2[HuggingFace]
* `text-embedding-ada-002` second generation embedding model for OpenAI
* models available through https://ai.azure.com/explore/models?selectedTask=embeddings[Azure AI Studio] or https://learn.microsoft.com/en-us/azure/ai-services/openai/concepts/models[Azure OpenAI]

> * `text-embedding-004` model for https://cloud.google.com/vertex-ai/generative-ai/docs/model-reference/text-embeddings-api#generative-ai-get-text-embedding-python_vertex_ai_sdk[Google Vertex AI]

Reviewer: Should we link to https://cloud.google.com/vertex-ai/generative-ai/docs/model-reference/text-embeddings-api instead of the python vertex ai specific section? The top of the page has a model overview

[[reindexing-data-infer]]
==== Ingest the data through the {infer} ingest pipeline

> Create the embeddings from the text by reindexing the data through the {infer} pipeline that uses the chosen model as the {infer} model.

Reviewer 2: ```suggestion
Create embeddings from the text by reindexing the data through the {infer} pipeline that uses your chosen model.
```

> This step - with using the {ref}/docs-reindex.html[reindex API] - simulates data ingestion through a pipeline.

Reviewer 2: ```suggestion
This step uses the {ref}/docs-reindex.html[reindex API] to simulate data ingestion through a pipeline.
```
